Sources/Swift/Tools/SentryLogBatcher.swift

```diff
@@ -6,14 +6,92 @@ import Foundation
 @_spi(Private) public class SentryLogBatcher: NSObject {
     
     private let client: SentryClient
+    private let flushTimeout: TimeInterval
+    private let maxBufferSize: Int
+    private let dispatchQueue: SentryDispatchQueueWrapper
     
-    @_spi(Private) public init(client: SentryClient) {
+    private var logBuffer: [SentryLog] = []
+    private let logBufferLock = NSLock()
+    private var currentFlushId: UUID?
+    
+    @_spi(Private) public init(
+        client: SentryClient,
+        flushTimeout: TimeInterval,
+        maxBufferSize: Int,
+        dispatchQueue: SentryDispatchQueueWrapper
+    ) {
         self.client = client
+        self.flushTimeout = flushTimeout
+        self.maxBufferSize = maxBufferSize
+        self.dispatchQueue = dispatchQueue
         super.init()
     }
     
+    @_spi(Private) public convenience init(client: SentryClient, dispatchQueue: SentryDispatchQueueWrapper) {
+        self.init(
+            client: client,
+            flushTimeout: 5,
+            maxBufferSize: 100,
+            dispatchQueue: dispatchQueue
+        )
+    }
+    
     func add(_ log: SentryLog) {
-        dispatch(logs: [log])
+        cancelFlush()
+
+        let shouldFlush = logBufferLock.synchronized {
+            logBuffer.append(log)
+            return logBuffer.count >= maxBufferSize
+        }
+        
+        if !shouldFlush {
+            scheduleFlush()
+        } else {
+            flush()
+        }
+    }
+    
+    @objc
+    public func flush() {
+        cancelFlush()
+
+        let logs = logBufferLock.synchronized {
+            let logs = Array(logBuffer)
+            logBuffer.removeAll()
+            return logs
+        }
+        
+        if !logs.isEmpty {
+            dispatch(logs: logs)
+        }
```

Comment:
`h`: When we exceed the `maxBufferSize` in `add` we call dispatch on the calling thread which calls then calls `encodeToJSONData`. So it can happen that we encode on the calling thread and block the main thread, which isn't ideal. Depending on the solution in https://github.com/getsentry/sentry-cocoa/pull/5628/files#r2207776056 this comment could get obsolete.